Components/FlightControl/RocketSM.cpp

```diff
@@ -348,10 +345,12 @@ RocketState Arm::HandleCommand(Command& cm)
     case CONTROL_ACTION: {
         switch (cm.GetTaskCommand()) {
         case RSC_POWER_TRANSITION_EXTERNAL:
-            //TODO: Transition to umbilical power - we should check to make sure umbilical power is available before doing so
+            PowerSelect::UmbilicalPower();
+            //TODO: we should check to make sure umbilical power is available before doing so
             break;
         case RSC_POWER_TRANSITION_ONBOARD:
-            //TODO: Transition to onboard power
+            PowerSelect::InternalPower();
+            //TODO: we should check to make sure internal power is available before doing so
             break;
         case RSC_GOTO_IGNITION:
             // Transition to ready for ignition state
```

Comment:
Got some more clarification on the ignition state transitions from Mo:
As soon as we enter the ignition state, we should start a timer for the igniter delay. If the timer expires before the launch command is received, we should go into Arm. If the launch command is received, we should wait until the timer expires before entering launch and opening the MEV